### PR TITLE
Feature/add i principal provider has value

### PR DESCRIPTION
**Pull request by moialbla**
Sometimes if you're working in asynchronous contexts and the developer is using this property to get the **IPrincipalProvide.Current** an exception will be thrown because HTTPContext is null,

### CODE REVIEW

File: src/Hosting/AspNetCore/Http/PrincipalProvider.cs [hunk 1]
```diff
@@ -13,5 +13,10 @@ public PrincipalProvider(IHttpContextAccessor accessor)
             this.accessor = accessor;
         }
         public IPrincipal Current => this.accessor.HttpContext.User;
```
Comment (sergiomcalzada) [inline src/Hosting/AspNetCore/Http/PrincipalProvider.cs:15]: We should also do a null check. If someone forgot to check the "HasValue" this will raise the exception
```
public IPrincipal Current => this.accessor.HttpContext?.User;
```

File: .github/workflows/stize.dotnet.CD.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: stize.dotnet.CD
```
Comment (sergiomcalzada) [inline .github/workflows/stize.dotnet.CD.yml:1]: Why is this file changed?

**Comment by moialbla** [comment]
@sergiomcalzada, I'm going to close this PR and create a new one, because I've created this feature from 0.0.2 instead of the main branch and that's why you see the .**yml** changes.
